### ISSUE TITLE
Edge count discrepancy with sparsifier

### ISSUE THREAD

**Issue by deklanw** [networkit/networkit#499]
I'm starting with a directed graph with 8541 edges. I run a Sparsifier with a ratio of 0.5. 

```python
    print(G.numberOfEdges(), localDegGraph.numberOfEdges())
```

```8541 4234```

So far so good. 

```python
    edge_c = 0

    def count_edges(source, target, _weight, _edge_id):
        nonlocal edge_c
        edge_c += 1

    localDegGraph.forEdges(count_edges)
    print(edge_c)
```

Gives:

```3972```

What's causing the discrepancy? I also ran a layout algorithm on `localDegGraph`, saved it with `writeGraphToGML` and also got 3972 edges.

**Comment by angriman** [comment]
Probably the sparsifier does not update edge count correctly. Which sparsifier are you using?

**Comment by deklanw** [comment]
@angriman 

Thanks for response. I'm using LocalDegreeSparsifier

**Comment by angriman** [comment]
`LocalDegree` sparsifier only computes edge scores, so the edges should be removed somewhere else. How exactly do you create `localDegGraph`?

**Comment by deklanw** [comment]
```python
# Initialize the algorithm
localDegSparsifier = nk.sparsification.LocalDegreeSparsifier()

# Get sparsified graph
localDegGraph = localDegSparsifier.getSparsifiedGraphOfSize(G, targetRatio)
G.numberOfEdges(), localDegGraph.numberOfEdges()
```
As in https://github.com/networkit/networkit/blob/Dev/notebooks/Sparsification.ipynb

**Comment by angriman** [comment]
Thank you. I ran this code:
```
import networkit as nk

G = nk.readGraph('./input/MIT8.edgelist', nk.Format.EdgeList, separator='\t', firstNode=0,
        continuous=False, directed=False)
G.indexEdges()

# Initialize the algorithm
localDegSparsifier = nk.sparsification.LocalDegreeSparsifier()

# Get sparsified graph
localDegGraph = localDegSparsifier.getSparsifiedGraphOfSize(G, .5)

edge_c = 0
def count_edges(source, target, _weight, _edge_id):
    global edge_c
    edge_c += 1

localDegGraph.forEdges(count_edges)
assert(localDegGraph.numberOfEdges() == edge_c)
```
but I could not replicate the issue, the edge count always match even for different ratios. Could you attach the network you are using so I can test it?

**Comment by deklanw** [comment]
My graph was directed, I believe if you test this out with a large directed graph you will find the problem

**Comment by deklanw** [comment]
Seems that `getSparsifiedGraphOfSize` is converting directed graphs to undirected

```python
    print(G.isDirected())
    print(sparsified_graph.isDirected())
```

```
True
False
```

**Comment by fabratu** [comment]
Closing this issue - problem is fixed in release 7.0.
